### PR TITLE
BI-9230 add LP & LLP journey feature flags

### PR DESCRIPTION
**Pull request by phudson-ch**
- Implement LP & LLP feature flags using Spring bean configuration
- Add FeatureOptions to CertificateOrderNotificationModel
- Address SonarQube issues [use EnumMap instead of HashMap]

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryLPFeatureFlagEnabledIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        // given
+        // when
+        CertificateOptionsMapper certificateOptionsMapper = mapperFactory.getCertificateOptionsMapper(CompanyType.LIMITED_LIABILITY_PARTNERSHIP);
+
+        // then
+        assertTrue(certificateOptionsMapper instanceof OtherCertificateOptionsMapper);
+    }
+
+    @Test
+    void returnLLPCertificateOptionsMapperWhenCompanyTypeLP() {
```
Comment (kinpang-CH) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryLPFeatureFlagEnabledIntegrationTest.java]: returnLPCertificateOptionsMapperWhenCompanyTypeLP
Change test name

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryLPFeatureFlagEnabledIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        // given
+        // when
+        CertificateOptionsMapper certificateOptionsMapper = mapperFactory.getCertificateOptionsMapper(TestConstants.LIMITED_COMPANY_TYPE);
+
+        // then
+        assertTrue(certificateOptionsMapper instanceof OtherCertificateOptionsMapper);
+    }
+
+    @Test
+    void returnLLPCertificateOptionsMapperWhenCompanyTypeLLP() {
```
Comment (kinpang-CH) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryLPFeatureFlagEnabledIntegrationTest.java]: returnOtherCertificateOptionsMapperWhenCompanyTypeLLP

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryLLPFeatureFlagEnabledIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        // given
+        // when
+        CertificateOptionsMapper certificateOptionsMapper = mapperFactory.getCertificateOptionsMapper(CompanyType.LIMITED_LIABILITY_PARTNERSHIP);
+
+        // then
+        assertTrue(certificateOptionsMapper instanceof LLPCertificateOptionsMapper);
+    }
+
+    @Test
+    void returnLLPCertificateOptionsMapperWhenCompanyTypeLP() {
```
Comment (kinpang-CH) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryLLPFeatureFlagEnabledIntegrationTest.java]: returnOtherCertificateOptionsMapperWhenCompanyTypeLLP

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryFeatureFlagsEnabledIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        // given
+        // when
+        CertificateOptionsMapper certificateOptionsMapper = mapperFactory.getCertificateOptionsMapper(CompanyType.LIMITED_LIABILITY_PARTNERSHIP);
+
+        // then
+        assertTrue(certificateOptionsMapper instanceof LLPCertificateOptionsMapper);
+    }
+
+    @Test
+    void returnLLPCertificateOptionsMapperWhenCompanyTypeLP() {
```
Comment (kinpang-CH) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryFeatureFlagsEnabledIntegrationTest.java]: returnLPCertificateOptionsMapperWhenCompanyTypeLLP

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryFeatureFlagsDisabledIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        // given
+        // when
+        CertificateOptionsMapper certificateOptionsMapper = mapperFactory.getCertificateOptionsMapper(TestConstants.LIMITED_COMPANY_TYPE);
+
+        // then
+        assertTrue(certificateOptionsMapper instanceof OtherCertificateOptionsMapper);
+    }
+
+    @Test
+    void returnLLPCertificateOptionsMapperWhenCompanyTypeLLP() {
```
Comment (kinpang-CH) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryFeatureFlagsDisabledIntegrationTest.java]: returnOtherCertificateOptionsMapperWhenCompanyTypeLLP

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryFeatureFlagsDisabledIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        // given
+        // when
+        CertificateOptionsMapper certificateOptionsMapper = mapperFactory.getCertificateOptionsMapper(CompanyType.LIMITED_LIABILITY_PARTNERSHIP);
+
+        // then
+        assertTrue(certificateOptionsMapper instanceof OtherCertificateOptionsMapper);
+    }
+
+    @Test
+    void returnLLPCertificateOptionsMapperWhenCompanyTypeLP() {
```
Comment (kinpang-CH) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryFeatureFlagsDisabledIntegrationTest.java]: returnOtherCertificateOptionsMapperWhenCompanyTypeLLP

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java [hunk 7]
```diff
@@ -45,7 +58,11 @@ void returnLLPCertificateOptionsMapperWhenCompanyTypeLLP() {
     @Test
     void returnLLPCertificateOptionsMapperWhenCompanyTypeLP() {
         // given
+        when(featureOptions.isLpCertificateOrdersEnabled()).thenReturn(true);
```
Comment (kinpang-CH) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java:61]: returnLPCertificateOptionsMapperWhenCompanyTypeLLP

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java [hunk 8]
```diff
@@ -35,7 +44,11 @@ void returnOtherCertificateOptionsMapperWhenCompanyTypeMatch() {
     @Test
     void returnLLPCertificateOptionsMapperWhenCompanyTypeLLP() {
         // given
+        when(featureOptions.isLpCertificateOrdersEnabled()).thenReturn(false);
+        when(featureOptions.isLlpCertificateOrdersEnabled()).thenReturn(true);
+
         // when
+        CertificateOptionsMapperFactory mapperFactory = new CertificateOptionsMapperFactory(llpCertificateOptionsMapper, lpCertificateOptionsMapper, otherCertificateOptionsMapper, featureOptions);
```
Comment (kinpang-CH) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java]: Should be moved to the // given block

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java [hunk 9]
```diff
@@ -1,31 +1,40 @@
...
+    }
 
     @Test
     void returnOtherCertificateOptionsMapperWhenCompanyTypeMatch() {
         // given
+        when(featureOptions.isLpCertificateOrdersEnabled()).thenReturn(true);
+        when(featureOptions.isLlpCertificateOrdersEnabled()).thenReturn(true);
+
         // when
+        CertificateOptionsMapperFactory mapperFactory = new CertificateOptionsMapperFactory(llpCertificateOptionsMapper, lpCertificateOptionsMapper, otherCertificateOptionsMapper, featureOptions);
```
Comment (kinpang-CH) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java]: Should be moved to the // given block

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java [hunk 10]
```diff
@@ -45,7 +58,11 @@ void returnLLPCertificateOptionsMapperWhenCompanyTypeLLP() {
     @Test
     void returnLLPCertificateOptionsMapperWhenCompanyTypeLP() {
         // given
+        when(featureOptions.isLpCertificateOrdersEnabled()).thenReturn(true);
+        when(featureOptions.isLlpCertificateOrdersEnabled()).thenReturn(false);
+
         // when
+        CertificateOptionsMapperFactory mapperFactory = new CertificateOptionsMapperFactory(llpCertificateOptionsMapper, lpCertificateOptionsMapper, otherCertificateOptionsMapper, featureOptions);
```
Comment (kinpang-CH) [inline src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java]: Should be moved to the // given block

**Review by kinpang-CH — CHANGES_REQUESTED** [review]
Address PR comments

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryLPFeatureFlagEnabledIntegrationTest.java (hunk 1 above)
Comment (phudson-ch) [reply src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryLPFeatureFlagEnabledIntegrationTest.java]: LLP changed to LP

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryLLPFeatureFlagEnabledIntegrationTest.java (hunk 3 above)
Comment (phudson-ch) [reply src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryLLPFeatureFlagEnabledIntegrationTest.java]: LLP changed to Other in method name

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryFeatureFlagsDisabledIntegrationTest.java (hunk 5 above)
Comment (phudson-ch) [reply src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryFeatureFlagsDisabledIntegrationTest.java]: LLP changed to Other in method name

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java (hunk 7 above)
Comment (phudson-ch) [reply src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java:61]: LLP changed to LP in method name

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java (hunk 8 above)
Comment (phudson-ch) [reply src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java]: Done

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java (hunk 9 above)
Comment (phudson-ch) [reply src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java]: done

File: src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java (hunk 10 above)
Comment (phudson-ch) [reply src/test/java/uk/gov/companieshouse/ordernotification/emailsendmodel/CertificateOptionsMapperFactoryTest.java]: done

**Review by dlgroves-ch — APPROVED** [review]

**Review by kinpang-CH — APPROVED** [review]

**Review by cgregoryCH — APPROVED** [review]

**Pull request merged**